tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/SetInRule.scala

```diff
@@ -120,11 +122,25 @@ class SetInRule(rewriter: SymbStateRewriter) extends RewritingRule {
       tla
         .or(tla.not(tla.apalacheSelectInSet(pair.toNameEx, relation.toNameEx)), tla.and(inDom.toNameEx, inCdm.toNameEx))
     }
-
     val relElems = nextState.arena.getHas(relation)
     rewriter.solverContext.assertGroundExpr(tla.equiv(pred.toNameEx, tla.and(relElems.map(onPair): _*)))
 
-    rewriter.rewriteUntilDone(nextState.setRex(pred.toNameEx))
+    val partialConstraintDomainState = rewriter.rewriteUntilDone(nextState.setRex(pred.toNameEx))
+
+    // S = DOMAIN f
+    val domainEx =
+      tla
+        .eql(
+            funsetDom.toNameEx,
+            tla.dom(funCell.toNameEx).as(domType),
+        )
+        .as(BoolT1())
+
+    rewriter.rewriteUntilDone(
+        partialConstraintDomainState.setRex(
+            tla.and(partialConstraintDomainState.ex, domainEx)
```

Comment:
Do we have to call the rewriter twice, or can we just do

```suggestion
    // S = DOMAIN f
    val domainEx =
      tla
        .eql(
            funsetDom.toNameEx,
            tla.dom(funCell.toNameEx).as(domType),
        )
        .as(BoolT1())

    rewriter.rewriteUntilDone(
        nextState.setRex(tla.and(pred.toNameEx, domainEx)
```

Reply:
True